### ISSUE TITLE
Dolphin doesn’t register input 

### ISSUE THREAD

**Issue by Mmgxhaze97** [dragoonDorise/EmuDeck#1335]
### Prerequisite Steps

- [X] I have searched existing issues
- [ ] This issue is not a duplicate of an existing one
- [X] I have checked the [EmuDeck Wiki](https://emudeck.github.io/)
- [X] I have read the [Troubleshooting Page](https://emudeck.github.io/troubleshooting/steamos/) on the EmuDeck Wiki

### What distro are you running?

SteamOS

### Which SteamOS Branch are you using?

Stable

### What happened?

I bought a new sd card and after migrating my emu deck set up I found that dolphin doesn’t register my inputs anymore. After doing all sorts of troubleshooting such as removing sd card, reinstalling emu deck and even reinstalling steam os and starting fresh no luck.

https://github.com/user-attachments/assets/4b238bf2-936e-4457-9951-6ddce6f08bdd

![IMG_4991](https://github.com/user-attachments/assets/fea9bebb-b9d9-402a-abd2-77ffa258d557)

┆Issue is synchronized with this [Trello card](https://trello.com/c/yLPJqwCe) by [Unito](https://www.unito.io)

**Comment by ThomasRumas** [comment]
I’ve got the same problem on my Steam Deck, and also on a Rog Ally. 

Installing, updating, reset configuration… don’t solve the issue. 

If I change to Steam Deck controller on Dolphin’s input configuration it works but not able to use hotkeys from ES-DE software…

**Comment by daniframos** [comment]
> I’ve got the same problem on my Steam Deck with Bazzite OS, and also on a Rog Ally with Windows.
> 
> Installing, updating, reset configuration… don’t solve the issue.
> 
> If I change to Steam Deck controller and not on Steam Virtual Input on Dolphin’s controller configuration it works but not able to use hotkeys from ES-DE software…

Facing the same situation. I've tried all the controller settings without success. It only works with the Steam controller in desktop mode.

**Comment by strauzen** [comment]
Same here, only after the last update. I tried to reset emudeck and it still didn't solve the problem.

**Comment by Laufzeitfehler9** [comment]
Same Problem jere since Last Update.

**Comment by Arc86** [comment]
I was able to get it working again in GameMode with the following steps: 

From the drop down menu:

- Select **SteamDeck/0/SteamDeck**
- Reassign all buttons 

Under options in same window: 

- Enable Always connected (you might need to scroll down on a steam deck screen)

Press close button and return to GameMode, it should now work.

**Comment by ThomasRumas** [comment]
> I was able to get it working again in GameMode with the following steps:
> 
> Go to controllers and from the drop down menu:
> 
> * Select **SteamDeck/0/SteamDeck**
> * Reassign all buttons
> 
> Under options in same window:
> 
> * Enable Always connected (you might need to scroll down on a steam deck screen)
> 
> Press close button and return to GameMode, it should now work.

Does it also keep hotkeys from ES-DE ?

**Comment by Laufzeitfehler9** [comment]
> > I was able to get it working again in GameMode with the following steps:
> > Go to controllers and from the drop down menu:
> > 
> > * Select **SteamDeck/0/SteamDeck**
> > * Reassign all buttons
> > 
> > Under options in same window:
> > 
> > * Enable Always connected (you might need to scroll down on a steam deck screen)
> > 
> > Press close button and return to GameMode, it should now work.
> 
> Does it also keep hotkeys from ES-DE ?

No it's only a workaround. Hotkeys only working with SteamInput. Eventually Steam changed some stuff with SteamInput in latest SteamOS Update?

**Comment by strauzen** [comment]
I've tried your solution but managed to get it working even better with the hotkeys and all.

I realized that the name of the SteamInput controller is different in desktop than gaming mode. 

So to get it working you just need to launch dolphin from game mode and switch the controller marked as disconnected for the one that appears as connected with Steam input in the name. You need to fix it for every profile and the hotkeys config.

Then everything works as expected, no need to remap any buttons.

If you need more help let me know I'm not at home and can't take some screenshots atm

**Comment by Laufzeitfehler9** [comment]
> I've tried your solution but managed to get it working even better with the hotkeys and all.
> 
> I realized that the name of the SteamInput controller is different in desktop than gaming mode.
> 
> So to get it working you just need to launch dolphin from game mode and switch the controller marked as disconnected for the one that appears as connected with Steam input in the name. You need to fix it for every profile and the hotkeys config.
> 
> Then everything works as expected, no need to remap any buttons.
> 
> If you need more help let me know I'm not at home and can't take some screenshots atm

Thank you for your Feedback.
For me there is no other controller available in dropdown menu named steam Input.

**Comment by Arc86** [comment]
I think I finally found the definitive answer how to get this configured pro

> I've tried your solution but managed to get it working even better with the hotkeys and all.
> 
> I realized that the name of the SteamInput controller is different in desktop than gaming mode.
> 
> So to get it working you just need to launch dolphin from game mode and switch the controller marked as disconnected for the one that appears as connected with Steam input in the name. You need to fix it for every profile and the hotkeys config.
> 
> Then everything works as expected, no need to remap any buttons.
> 
> If you need more help let me know I'm not at home and can't take some screenshots atm

Yeah, you are right. Seems like in game mode it's using the following controller that is not visible when in Desktop Mode: 

**SteamDeck/0/SteamDeck Controller** 

When you go into your user folder you can edit the config/profile files manually to this controller. After this everything will work as expected (at least for me) 

Change the following line starting with Device = to: 
Device = "SteamDeck/0/SteamDeck Controller"

The files I've changed are: 

Home/.var/app/org.DolphinEmu.dolphin-emu/config/profiles/GCPad

- GC_base.ini
- GC_nintendo_layout.ini
- GC_xbox_layout.ini

Home/.var/app/org.DolphinEmu.dolphin-emu/config

- GCPadNew.ini
- Hotkeys.ini

I don't play wii games, but you might need to do the same for the Wii profiles.
Go back to Dolphin and load one of the changed GameCube profiles to be sure that it selects the proper controller device.

**Comment by Laufzeitfehler9** [comment]
> > I've tried your solution but managed to get it working even better with the hotkeys and all.
> > I realized that the name of the SteamInput controller is different in desktop than gaming mode.
> > So to get it working you just need to launch dolphin from game mode and switch the controller marked as disconnected for the one that appears as connected with Steam input in the name. You need to fix it for every profile and the hotkeys config.
> > Then everything works as expected, no need to remap any buttons.
> > If you need more help let me know I'm not at home and can't take some screenshots atm
> 
> Yeah, you are right. Seems like in game mode it's using the following controller that is not visible when in Desktop Mode:
> 
> **SteamDeck/0/SteamDeck Controller**
> 
> When you go into your user folder you can edit the config/profile files manually to this controller. After this everything will work as expected (at least for me)
> 
> Change the following line starting with Device = to: Device = "SteamDeck/0/SteamDeck Controller"
> 
> The files I've changed are:
> 
> Home/.var/app/org.DolphinEmu.dolphin-emu/config/profiles/GCPad
> 
> * GC_base.ini
> * GC_nintendo_layout.ini
> * GC_xbox_layout.ini
> 
> Home/.var/app/org.DolphinEmu.dolphin-emu/config
> 
> * GCPadNew.ini
> * Hotkeys.ini
> 
> I don't play wii games, but you might need to do the same for the Wii profiles. Go back to Dolphin and load one of the changed GameCube profiles to be sure that it selects the proper controller device.

Thank you, i will try this later.

Is it possible for you to try also a external controller with this config? I think the missing virtual Input is a placeholder from Valve for the SteamInput. I mean this is the config where SteamInput is taking control with his own overlay. In this case all Controllers are compatible with Dolphin not only the builtin-one.

**Comment by Arc86** [comment]
The external controller (Xbox controller connected through Bluetooth) is working for me when I reorder the controls in the Steam Deck with the Xbox Controller as the primary controller. 

I expect to get it fully working with multiple controllers we need to start adding 

SteamDeck/1/SteamDeck Controller
SteamDeck/2/SteamDeck Controller
SteamDeck/3/SteamDeck Controller

In the appropriate places in the config files, however, I've not tested this.
